src/mgmt/srv_drpc.c

```diff
@@ -572,6 +572,134 @@ ds_mgmt_drpc_pool_destroy(Drpc__Call *drpc_req, Drpc__Response *drpc_resp)
 	mgmt__pool_destroy_req__free_unpacked(req, NULL);
 }
 
+void ds_mgmt_drpc_pool_set_prop(Drpc__Call *drpc_req, Drpc__Response *drpc_resp)
+{
+	Mgmt__PoolSetPropReq *req         = NULL;
+	Mgmt__PoolSetPropResp resp        = MGMT__POOL_SET_PROP_RESP__INIT;
+	daos_prop_t          *new_prop    = NULL;
+	daos_prop_t          *result      = NULL;
+	char                 *out_str_val = NULL;
+	uuid_t                uuid;
+	uint8_t              *body;
+	size_t                len;
+	int                   rc;
+
+	/* Unpack the inner request from the drpc call body */
+	req = mgmt__pool_set_prop_req__unpack(NULL, drpc_req->body.len,
+					     drpc_req->body.data);
+
+	if (req == NULL) {
+		drpc_resp->status = DRPC__STATUS__FAILED_UNMARSHAL_PAYLOAD;
+		D_ERROR("Failed to unpack req (pool setprop)\n");
+		return;
+	}
+
+	D_INFO("Received request to set pool property on %s\n", req->uuid);
+
+	rc = uuid_parse(req->uuid, uuid);
+	if (rc != 0) {
+		D_ERROR("Couldn't parse '%s' to UUID\n", req->uuid);
+		D_GOTO(out, rc = -DER_INVAL);
+	}
+
+	new_prop = daos_prop_alloc(1);
+	if (new_prop == NULL) {
+		D_ERROR("Failed to allocate daos property\n");
+		D_GOTO(out, rc = -DER_NOMEM);
+	}
+
+	if (req->property_case != MGMT__POOL_SET_PROP_REQ__PROPERTY_NUMBER) {
+		D_ERROR("Pool property request must be numeric\n");
+		D_GOTO(out, rc = -DER_INVAL);
+	}
+	new_prop->dpp_entries[0].dpe_type = req->number;
+
+	switch (req->value_case) {
+	case MGMT__POOL_SET_PROP_REQ__VALUE_STRVAL:
+		D_ASPRINTF(out_str_val, "%s", req->strval);
+		if (out_str_val == NULL) {
+			D_GOTO(out, rc = -DER_NOMEM);
+		}
```

Comment:
nit - Don't bother changing it unless you have to change something else, but in the C code the style has been to avoid curly brackets for single-line if statements.

Reply:
Done